lucet-runtime/lucet-runtime-internals/src/future.rs

```diff
@@ -150,91 +122,120 @@ impl InstanceHandle {
     /// runtime between async future yields (invocations of `.poll()` on the
     /// underlying generated future) if `runtime_bound` is provided. This
     /// behaves the same way as `Instance::run_async()`.
-    pub async fn run_async_start<'a>(
-        &'a mut self,
-        runtime_bound: Option<u64>,
-    ) -> Result<(), Error> {
-        if !self.is_not_started() {
-            return Err(Error::StartAlreadyRun);
-        }
-        let start = match self.module.get_start_func()? {
-            Some(start) => start,
-            None => return Ok(()),
+    pub fn run_async_start<'a>(&'a mut self) -> RunAsync<'a> {
```

Comment:
Hmm, I mean it's just `run_async("entrypoint").bound_inst_count(count).await`, rather than `run_async("entrypoint", Some(4)).await`. Shouldn't make things much more fiddly, and, in my opinion, that's a nicer API than having to pass through an optional (closer to a builder-style API). Especially if other runtime options are added in the future, in my opinion, this API is much easier to use.

Reply:
Caveat: for the `Box::pin`/custom executor tests, the API usage is indeed more fiddly. That's because calling `Box::pin` on a mutable reference that isn't assigned to a variable causes the lifetime to go out of scope. `.await` does not have the same issue.

The only place this would be fiddly would be if the future returned by `run_async` needs to be moved - something like `tokio::spawn` or similar. But, notably, `run_async` requires a `&mut self` reference, which (if my understanding is correct) limits the lifetime and prevents usage (if my understanding is incorrect, at least prevents usage for any reasonable usecase) with `tokio::spawn` without a custom async block anyways.